client/src/search.tsx

```diff
@@ -106,16 +106,22 @@ function HighlightMatch({ title, q }) {
   );
 }
 
-function BreadcrumbURI({ uri, substrings }) {
-  if (substrings) {
+function BreadcrumbURI({
+  uri,
+  positions,
+}: {
+  uri: string;
+  positions: number[];
+}) {
+  if (positions) {
+    const chars = uri.split("");
     return (
       <small>
-        {substrings.map((part, i) => {
-          const key = `${part.str}:${i}`;
-          if (part.match) {
-            return <mark key={key}>{part.str}</mark>;
+        {chars.map((char, i) => {
+          if (positions.includes(i)) {
+            return <mark key={i}>{char}</mark>;
```

Comment:
This improves the previous `Substring[]` iteration logic by quite a bit! Its much more straightforward using the name `positions` which will be an array of character numeric indexes (positions). And iterating over the array of characters in `chars` to then check if the positions array includes the index. This is much easier to comprehend than the previous logic.

Really nice work here!